### ISSUE TITLE
Code generation failing when JPA Entity are directly included with spring dependency in it

### ISSUE THREAD

**Issue by poovamraj** [jOOQ/jOOQ#10298]
### Expected behavior and actual behavior: Instead of code being generated I am getting exception at 

```
org.jooq.exception.DataAccessException: Error while exporting schema
	at org.jooq.meta.extensions.AbstractInterpretingDatabase.connection(AbstractInterpretingDatabase.java:103)
	at org.jooq.meta.extensions.AbstractInterpretingDatabase.create0(AbstractInterpretingDatabase.java:77)
	at org.jooq.meta.AbstractDatabase.create(AbstractDatabase.java:274)
	at org.jooq.meta.AbstractDatabase.create(AbstractDatabase.java:263)
	at org.jooq.meta.AbstractDatabase.setConnection(AbstractDatabase.java:253)
	at org.jooq.codegen.GenerationTool.run0(GenerationTool.java:521)
	at org.jooq.codegen.GenerationTool.run(GenerationTool.java:222)
	at org.jooq.codegen.GenerationTool.generate(GenerationTool.java:217)
	at org.jooq.codegen.GenerationTool.main(GenerationTool.java:189)
Caused by: org.springframework.beans.factory.BeanDefinitionStoreException: Failed to read candidate component class: URL [jar:file:/home/local/ZOHOCORP/poovam-5255/.gradle/caches/modules-2/files-2.1/org.springframework.boot/spring-boot-autoconfigure/2.1.7.RELEASE/2c9d3e2c6ea3cb435e99e2973009636b62a9d816/spring-boot-autoconfigure-2.1.7.RELEASE.jar!/org/springframework/boot/autoconfigure/web/servlet/DispatcherServletAutoConfiguration$DispatcherServletConfiguration.class]; nested exception is java.lang.IllegalStateException: Error processing condition on org.springframework.boot.autoconfigure.web.servlet.DispatcherServletAutoConfiguration$DispatcherServletConfiguration
	at org.springframework.context.annotation.ClassPathScanningCandidateComponentProvider.scanCandidateComponents(ClassPathScanningCandidateComponentProvider.java:454)
	at org.springframework.context.annotation.ClassPathScanningCandidateComponentProvider.findCandidateComponents(ClassPathScanningCandidateComponentProvider.java:316)
	at org.jooq.meta.extensions.jpa.JPADatabase.export(JPADatabase.java:146)
	at org.jooq.meta.extensions.AbstractInterpretingDatabase.connection(AbstractInterpretingDatabase.java:100)
	... 8 common frames omitted
Caused by: java.lang.IllegalStateException: Error processing condition on org.springframework.boot.autoconfigure.web.servlet.DispatcherServletAutoConfiguration$DispatcherServletConfiguration
	at org.springframework.boot.autoconfigure.condition.SpringBootCondition.matches(SpringBootCondition.java:60)
	at org.springframework.context.annotation.ConditionEvaluator.shouldSkip(ConditionEvaluator.java:108)
	at org.springframework.context.annotation.ConditionEvaluator.shouldSkip(ConditionEvaluator.java:88)
	at org.springframework.context.annotation.ConditionEvaluator.shouldSkip(ConditionEvaluator.java:71)
	at org.springframework.context.annotation.ClassPathScanningCandidateComponentProvider.isConditionMatch(ClassPathScanningCandidateComponentProvider.java:515)
	at org.springframework.context.annotation.ClassPathScanningCandidateComponentProvider.isCandidateComponent(ClassPathScanningCandidateComponentProvider.java:498)
	at org.springframework.context.annotation.ClassPathScanningCandidateComponentProvider.scanCandidateComponents(ClassPathScanningCandidateComponentProvider.java:431)
	... 11 common frames omitted
Caused by: java.lang.NullPointerException: null
	at org.springframework.boot.autoconfigure.web.servlet.DispatcherServletAutoConfiguration$DefaultDispatcherServletCondition.getMatchOutcome(DispatcherServletAutoConfiguration.java:159)
	at org.springframework.boot.autoconfigure.condition.SpringBootCondition.matches(SpringBootCondition.java:47)
	... 17 common frames omitted
```

> Task :generateProjectJooqSchemaSource FAILED

FAILURE: Build failed with an exception.

I am using https://github.com/etiennestuder/gradle-jooq-plugin 

My configuration for the plugin 

```
dependencies {
    jooqRuntime("org.jooq:jooq-meta-extensions")
    jooqRuntime(files(sourceSets.main.output.classesDirs))
}

jooq {
    project(sourceSets.main) {
        generator {
            database {
                name = 'org.jooq.meta.extensions.jpa.JPADatabase'
                includes = ".*"
            }
        }
    }
}
```

It does work when the module only contains entities and no other dependency but this structure is not suitable for my project.

Great work you have done here, Just wanted to know whether I could make it work for my use case

### Versions:

```
- jOOQ:
|    +--- org.springframework.boot:spring-boot-starter-jooq -> 2.1.7.RELEASE
|    |    +--- org.springframework.boot:spring-boot-starter-jdbc:2.1.7.RELEASE (*)
|    |    +--- org.springframework:spring-tx:5.1.9.RELEASE (*)
|    |    \--- org.jooq:jooq:3.11.11 -> 3.13.1
|    |         +--- org.reactivestreams:reactive-streams:1.0.2
|    |         \--- javax.xml.bind:jaxb-api:2.3.1 (*)
```

- Java: 11
- Database (include vendor): JPADatabase
- OS: Ubuntu
- JDBC Driver (include name if inofficial driver): -

**Comment by lukaseder** [comment]
Thanks for your report.

Can you show an example of such an entity? Or even better, if you have some time, provide an MCVE (https://github.com/jOOQ/jOOQ-mcve), which helps reproduce this case?

**Comment by poovamraj** [comment]
I had quite a few entities. A simple one is

```
@Entity
public class A {

     @Id
     int id;
}
```

I will try to share an MCVE. In the meantime, for anyone looking for help, I added the Liquibase generation to overcome this. But the reason for adding Liquibase was for the sole purpose of JOOQ and in hopes, it will be useful in future when my project becomes bigger

I also think generating DDL (option provided in hibernate) would also work (but when I tried, the DDL was generated without `;`). I am not using the DDL method since it will be generated on runtime and I wanted a compile-time solution.

**Comment by lukaseder** [comment]
I see, thanks for the additional explanation. That seems to be a duplicate of https://github.com/jOOQ/jOOQ/issues/10268, then.

> (but when I tried, the DDL was generated without ;)

That will be supported soon: https://github.com/jOOQ/jOOQ/issues/10164

**Comment by lukaseder** [comment]
Issue #10268 has been implemented for jOOQ 3.14. In the meantime, the workaround is documented here: https://github.com/jOOQ/jOOQ/issues/10298#issuecomment-647445774
